File: pattern/LinkedList In-place Reversal/61_Rotate_List/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(k ==0 || head == null || head.next == null){
            return head;
        }
        ListNode dummy = head;
        int n = 1 ;
        while(dummy.next != null){
            dummy = dummy.next;
            n++; 
        }
        k = k%n;
        if(k == 0){
            return head;
        }
        ListNode current = head;
        for(int i = 0 ; i < n - k - 1 ; i++){
            current = current.next;
        }
        ListNode newHead = current.next;
        current.next = null;
        dummy.next = head;
        return newHead;

    }
}

File: pattern/LinkedList In-place Reversal/61_Rotate_List/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:

        if not head or not head.next or  k==0:
            return head
        dummy = head;
        l = 1;
        while dummy.next:
            dummy = dummy.next
            l=l+1
        k = k%l
        print(k)
        if k==0:
            return head
        current = head
        for i in range (0 , l-k-1):
            current = current.next
        newHead = current.next
        current.next =None
        dummy.next = head
        return newHead